pkg/status/conditions.go

```diff
@@ -0,0 +1,32 @@
+package status
+
+import (
+	"github.com/kyma-project/lifecycle-manager/api/v1beta1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetConditions checks if all required conditions do exists and removes deprecated Conditions.
+// If one or many Conditions are missing, they will be created and `true` will be returned
+// to determine that an update is needed.
+// If all required conditions exist, they will be set to `False`.
+func SetConditions(kyma *v1beta1.Kyma, watcherEnabled bool) bool {
+	conditionUpdateNeeded := false
+
+	existingConditions := kyma.Status.Conditions
+	// Remove deprecated `Ready` condition
+	if deprecatedCondition := apimeta.FindStatusCondition(existingConditions,
```

Comment:
please add a TODO here to indicate remove this tech debt code later, and do we have a backlog issue related to this? If no, please create one.

Reply:
consider rename this `DeprecatedConditionReady` to `DeprecatedConditionTypeReady`